lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -662,6 +662,34 @@ impl SegmentEntry for ProxySegment {
         Ok(read_points)
     }
 
+    fn read_random_filtered<'a>(
```

Comment:
It is very similar to the other two read implementations. I wonder if it would make sense to merge them.

I assume you didn't merge it with `read_ordered_filtered` using a 'Random` order variant because this one is not ordered nor deterministic at all?

Reply:
They are pretty similar, yes, but the assumptions are different. I don't discard the possibility to merge them, though